tests/unit/sys/test_output.cxx

```diff
@@ -125,16 +126,184 @@ TEST_F(OutputTest, DisableEnableStdout) {
   std::remove(filename);
 }
 
-TEST_F(OutputTest, CleanupAndGetInstance) {
+TEST_F(OutputTest, GetInstance) {
   Output *local_output = Output::getInstance();
-  EXPECT_NE(local_output, nullptr);
 
-  local_output->cleanup();
+  EXPECT_NE(local_output, nullptr);
 
-  // Get a new instance
   Output *new_output = Output::getInstance();
 
-  *new_output << "Hello, world!\n";
+  EXPECT_EQ(local_output, new_output);
+}
+
+TEST_F(OutputTest, ConditionalGetBase) {
+  Output local_output_base;
+  ConditionalOutput local_output(&local_output_base);
 
-  EXPECT_EQ(buffer.str(), "Hello, world!\n");
+  EXPECT_EQ(local_output.getBase(), &local_output_base);
+}
+
+TEST_F(OutputTest, ConditionalCheckIsEnabled) {
+  Output local_output_base;
+  ConditionalOutput local_output(&local_output_base);
+
+  EXPECT_TRUE(local_output.isEnabled());
+  local_output.disable();
+  EXPECT_FALSE(local_output.isEnabled());
+  local_output.enable();
+  EXPECT_TRUE(local_output.isEnabled());
+  local_output.enable(false);
+  EXPECT_FALSE(local_output.isEnabled());
+  local_output.enable(true);
+  EXPECT_TRUE(local_output.isEnabled());
+}
+
+TEST_F(OutputTest, ConditionalJustStdOutCpp) {
+  Output local_output_base;
+  ConditionalOutput local_output(&local_output_base);
+
+  local_output << "Hello, world!" << 1 << "\n";
```

Comment:
`"\n"` could be replaced by `std::endl` to make sure that is working as well.

Reply:
Oh, to check those overloads. Good catch, thanks.